### DISCUSSION TITLE
ImportError: cannot import name ‘mtext’ from ‘matplotlib.axes’

### DISCUSSION THREAD

**Discussion by Hkgayau** [CommonRoad/commonroad-io#5] [Q&A] [answered]
I was trying to follow the "getting-started" tutorial and encounter this error.
```python
from commonroad.common.file_reader import CommonRoadFileReader
from commonroad.visualization.mp_renderer import MPRenderer



file_path = "location4.xml"

scenario, planning_problem_set = CommonRoadFileReader(file_path).open()

# plot the scenario
rnd = MPRenderer(figsize=(25, 10))
scenario.draw(rnd)
rnd.render(show=True)
```

```
Traceback (most recent call last):
  File "C:\Users\Sylas Lo\Study\Sem 8\Bachelorarbeit\test\cmnRd.py", line 6, in <module>
    from commonroad.visualization.mp_renderer import MPDrawParams
  File "C:\Users\Sylas Lo\Study\Sem 8\Bachelorarbeit\test\.venv\Lib\site-packages\commonroad\visualization\mp_renderer.py", line 68, in <module>
    from commonroad.visualization.traffic_sign import draw_traffic_light_signs
  File "C:\Users\Sylas Lo\Study\Sem 8\Bachelorarbeit\test\.venv\Lib\site-packages\commonroad\visualization\traffic_sign.py", line 8, in <module>
    from matplotlib.axes import Axes, mtext
test\.venv\Lib\site-packages\matplotlib\axes\__init__.py)
(.venv) PS C:\Users\Sylas Lo\Study\Sem 8\Bachelorarbeit\test> python .\cmnRd.py
Traceback (most recent call last):
  File "C:\Users\Sylas Lo\Study\Sem 8\Bachelorarbeit\test\cmnRd.py", line 6, in <module>
    from commonroad.visualization.mp_renderer import MPRenderer
  File "C:\Users\Sylas Lo\Study\Sem 8\Bachelorarbeit\test\.venv\Lib\site-packages\commonroad\visualization\mp_renderer.py", line 68, in <module>
    from commonroad.visualization.traffic_sign import draw_traffic_light_signs
  File "C:\Users\Sylas Lo\Study\Sem 8\Bachelorarbeit\test\.venv\Lib\site-packages\commonroad\visualization\traffic_sign.py", line 8, in <module>
    from matplotlib.axes import Axes, mtext
ImportError: cannot import name 'mtext' from 'matplotlib.axes' (C:\Users\Sylas Lo\Study\Sem 8\Bachelorarbeit\test\.venv\Lib\site-packages\matplotlib\axes\__init__.py)
```
I think I have everything correctly installed and I didn't touch any file in my site packages
------------------------- -----------
antlr4-python3-runtime    4.9.3
colorama                  0.4.6
commonroad-io             2024.1
commonroad-vehicle-models 3.0.2
contourpy                 1.2.1
cycler                    0.12.1
fonttools                 4.51.0
iso3166                   2.1.1
kiwisolver                1.4.5
lxml                      5.2.2
matplotlib                3.9.0
networkx                  3.3
numpy                     1.26.4
omegaconf                 2.3.0
packaging                 24.0
pillow                    10.3.0
pip                       24.0
protobuf                  3.20.2
pyparsing                 3.1.2
python-dateutil           2.9.0.post0
PyYAML                    6.0.1
Rtree                     1.2.0
scipy                     1.13.0
shapely                   2.0.4
six                       1.16.0
tqdm                      4.66.4

How can I resolve this issue?

**Comment by smaierhofer** [comment, marked as answer]
Please try to downgrade matplotlib to a version < 3.9
There seem to be incompatibilities with the latest major matplotlib version.
We will fix this in the next CommonRoad-IO release.
See [here](https://github.com/CommonRoad/commonroad-io/issues/4) for the corresponding issue.

**Reply by Hkgayau** [reply]
thank you, I also just realized matplotlib has released 3.9 a few days ago. I have downgraded to 3.8.3 and it works fine now!
